core/src/main/java/org/jruby/internal/runtime/AbstractIRMethod.java

```diff
@@ -32,8 +34,8 @@
     protected final IRScope method;
     protected final StaticScope staticScope;
     protected InterpreterContext interpreterContext = null;
-    protected int callCount = 0;
-    private MethodData methodData;
+    protected volatile int callCount = 0;
+    private transient MethodData methodData;
```

Comment:
Is the volatile here for disabling? Since our count is always checking `> threshold` I never bothered to make this volatile since someone will eventually cross the line. If it's volatile it will force a volatile read for every call.

And why transient?

Reply:
transient just a marker that its a cached field. 
method pbly can't serialize state due the IR scope but if it did this field is not needed